openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTest.java

```diff
@@ -236,4 +249,126 @@ private boolean compareDoubles(final double a, final double b, final int exclude
         );
     }
 
+    /**
+     * Compares files with the same name across two folders.
+     * If there are any differences in the models within such folders they are reported
+     * in a colored report. Requires python3 with termcolor installed for best readability.
+     *
+     * @param referenceModelsFolder Folder with reference models
+     * @param newModelsFolder       Folder with new models to compare to reference
+     * @return true if there are no differences and all the dependencies are installed, false otherwise.
+     * @throws IOException          In case the process errors.
+     * @throws InterruptedException In case awaiting for the process to finish fails.
+     * @since 1.0.19
+     */
+    private boolean compareModelFilesAndDoPrettyReport(
+            final String referenceModelsFolder,
+            final String newModelsFolder) throws IOException, InterruptedException {
+
+        // Check for differences and report in detailed manner!
+        final Process p = new ProcessBuilder(
+                "python",
+                "src/test/resources/diff_models.py",
+                referenceModelsFolder,
+                newModelsFolder
+        ).start();
+
+        // Fetch process output
+        try (final BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+            final StringBuilder builder = new StringBuilder();
+            String line = null;
+            while ((line = reader.readLine()) != null) {
+                builder.append(line);
+                builder.append(System.getProperty("line.separator"));
+            }
+            final String processOutput = builder.toString();
+
+            p.waitFor();
+            logger.error(processOutput);
+
+            return (p.exitValue() == 0);
+        }
+    }
+
+    /**
+     * Asserts that the two files have the same content.
+     *
+     * @param name      name of the file to compare for the assert message.
```

Comment:
Descriptions should start with an uppercase character, and finish with a final mark

Reply:
similar in other descriptions of fields, throw exceptions, ...